AB_TESTING.md

```diff
@@ -0,0 +1,47 @@
+# A/B Testing
+
+## Intro
+
+Pathfora allows the user to perform A/B testing with the widgets.
+
+___
+
+Example file: `./examples/AB-testing example.html`
+
+> In the example file, we create two widgets, each in a separate testing group (groups are indexed starting from 0). As we open the page, a random value is generated into browser's cookies. According to this value, proper widgets are rendered on the page.
+
+___
+
+Pathfora provides currently two types of A/B testing:
+
+## Testing Modes
+
+### `100`
+
+This default mode renders all widgets regardless the user visiting the page.
+
+### `50/50`
+
+This mode divides users into two groups, each group with their separate set of widgets on the page.
+
+## Usage 
+
+In order to initialize A/B testing on the website, `setABTestingMode( mode )` has to be called before widgets are created on the page. For example:
+
+```js
+window.pathfora.setABTestingMode('50/50'); // Initialize a 50/50 AB testing
```

Comment:
Would it make sense / be possible to set the testing mode globally or per group? I would say its not super likely to start but we definitely could run into users running multiple widgets on a single page. If that occurs it looks like we can only have one a/b testing type?

This might be moot if running a `50/50` campaign on a group with only one widget would result in a `100` a/b test? since there is only one to show?